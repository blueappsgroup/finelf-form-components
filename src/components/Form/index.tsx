import React, { FC, ReactElement, useMemo, useState } from 'react'
import { Formik, Form } from 'formik'
import styled from 'styled-components'

import { ThemeProvider } from '../../consts/theme'
import { device } from '../../consts/sizes'
import { FormProps, FormValuesType } from '../../types'
import RedirectPage from '../RedirectPage'
import {
  FormContext,
  getFormValuesFromCache,
  resetFormValueCache,
  handleSendDataToApi,
  getFieldsValuesFromUrl,
} from '../../utils'
import { formStatuses } from '../../consts/form'

const StyledForm = styled(Form)`
  display: flex;
  flex-direction: column;
  max-width: ${(props): string => props.theme.formMaxWidth};
  justify-self: center;
  background: ${(props): string => props.theme.formBgColor};
  font-family: ${(props): string => props.theme.fontFamily};
  border-radius: 6px;
  box-shadow: ${(props): string => props.theme.formBoxShadow};
`

const FormWrapper: FC<FormProps> = ({
  children,
  onSubmit,
  customTheme,
  id,
  stepsLength,
  stepsTitles,
  hasRedirect,
  redirectUrl,
  timeToRedirect,
  redirectHeaderText,
  logoImg,
  redirectMainImg,
  redirectBgImg,
  sendDataToApi,
  apiUrl,
  transactionName,
  propertyNamesFromUrl,
  dataWithUserAgent,
}) => {
  const trasationIdValue =
    transactionName &&
    new URLSearchParams(window.location.search).get(transactionName)
  const intialValuesFromUrl = useMemo(
    () =>
      (propertyNamesFromUrl &&
        propertyNamesFromUrl.length > 0 &&
        getFieldsValuesFromUrl(propertyNamesFromUrl)) ||
      {},
    [propertyNamesFromUrl]
  )
  const [redirectUrlPath, setRedirectUrlPath] = useState(redirectUrl)
  const [initialValues, setInitialValues] = useState({
    ...intialValuesFromUrl,
    ...getFormValuesFromCache(id),
    // eslint-disable-next-line @typescript-eslint/camelcase
    trasaction_id: trasationIdValue,
  })
  const [currentStep, setCurrentStep] = useState(0)
  const [fieldsForSkip, setFieldsForSkip] = useState<string[]>([])
  const [errorFromApi, setErrorFromApi] = useState<boolean>(false)
  const shouldRedirect = !errorFromApi && hasRedirect
  const addFieldForSkip = (key: string): void =>
    setFieldsForSkip([...fieldsForSkip, key])

  const handleSubmit = async (
    values: FormValuesType,
    props: {
      resetForm: () => void
      setStatus: (status: string) => void
    }
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ): Promise<any> => {
    if (onSubmit) {
      onSubmit(values, props)
    }

    try {
      if (sendDataToApi && apiUrl) {
        const response = await handleSendDataToApi(
          values,
          apiUrl,
          id,
          fieldsForSkip,
          dataWithUserAgent
        )
        const {
          redirectUrl: urlFromApi,
          status: statusFromApi,
        } = await response.json()

        if (statusFromApi === false) {
          setErrorFromApi(true)
          return
        }

        urlFromApi && setRedirectUrlPath(urlFromApi)
      }

      props.resetForm()
      props.setStatus(formStatuses.submited)
    } catch (e) {
      props.setStatus(formStatuses.error)
    }
  }
  const handleReset = (): void => {
    resetFormValueCache(id)
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    setInitialValues({} as any)
  }

  const prevStep: Function = () => setCurrentStep(currentStep - 1)

  const nextStep: Function = () => setCurrentStep(currentStep + 1)

  return (
    <FormContext.Provider
      value={{
        id,
        apiUrl,
        stepsLength,
        currentStep,
        stepsTitleList: stepsTitles,
        nextStep,
        prevStep,
        fieldsForSkip,
        addFieldForSkip,
        errorFromApi,
      }}
    >
      <ThemeProvider customTheme={{ ...customTheme }}>
        <Formik
          enableReinitialize
          initialValues={initialValues}
          onSubmit={handleSubmit}
          onReset={handleReset}
        >
          {(props): ReactElement =>
            (shouldRedirect && props.status === formStatuses.submited && (
              <RedirectPage
                redirectUrl={redirectUrlPath}
                backgroundImage={redirectBgImg}
                logoImg={logoImg}
                headerText={redirectHeaderText}
                timeToRedirect={timeToRedirect}
                mainImg={redirectMainImg}
              />
            )) || (
              <StyledForm id={id}>
                {children}
              </StyledForm>
            )
          }
        </Formik>
      </ThemeProvider>
    </FormContext.Provider>
  )
}

export default FormWrapper
