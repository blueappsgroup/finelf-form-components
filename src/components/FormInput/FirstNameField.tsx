import React, { FC } from 'react'
import { Field } from 'formik'

import { TextFieldProps } from '../../types'
import BaseField from './base'
import { validateText } from './validate'

const FirstNameField: FC<TextFieldProps> = ({
  id,
  label = 'ImiÄ™',
  placeholder,
  name,
  required = false,
  validate = validateText,
  showError = true,
}) => (
  <Field
    id={id}
    label={label}
    placeholder={placeholder}
    component={BaseField}
    required={required}
    showError={showError}
    name={name}
    validate={validate(required)}
  />
)

FirstNameField.defaultProps = {
  name: 'first_name',
}

export default FirstNameField
