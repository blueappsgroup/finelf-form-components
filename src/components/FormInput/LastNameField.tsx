import React, { ReactElement } from 'react'
import { TextFieldProps } from '../../types'
import TextField from './TextField'

const LastNameField: (props: TextFieldProps) => ReactElement = ({
  id,
  label = 'Nazwisko',
  placeholder = 'Nazwisko',
  name = 'last_name',
  required = true,
  validate,
  showError = true,
}) => (
  <TextField
    id={id}
    label={label}
    placeholder={placeholder}
    required={required}
    showError={showError}
    name={name}
    validate={validate}
  />
)

export default LastNameField
