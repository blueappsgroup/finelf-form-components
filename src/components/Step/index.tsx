import React, {
  FC,
  useContext,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react'
import styled from 'styled-components'
import { useFormikContext } from 'formik'

import Button from '../Button'
import StepHeader from './StepHeader'
import { FormContext } from '../../utils'
import { device } from '../../consts/sizes'

const Wrapper = styled.div<{ visible: boolean }>`
  display: ${(props): string => (props.visible ? 'block' : 'none')};
`

type ButtonWrapperTypes = { isFirstStep: boolean }

// prettier-ignore
const ButtonsWrapper = styled.div<ButtonWrapperTypes>`
  display: flex;
  flex-direction: column;
  
  & button {
    flex-grow: 1;
  }

  @media ${device.mobileL} {
    flex-direction: row;

    & button {
      &:first-of-type {
        margin-right: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '15px'};
      }

      &:last-child {
        margin-left: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '15px'};
      }
    }
  }

  @media ${device.tablet} {
    & button {
      &:first-of-type {
        margin-right: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '30px'};
      }

      &:last-child {
        margin-left: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '30px'};
      }
    }
  }
`

const StepHeaderWrapper = styled.div`
  display: flex;
`

type Props = {
  stepIndex: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: any
  onStepComplete?: () => void
}

const Step: FC<Props> = ({ children, stepIndex, onStepComplete }) => {
  const wrapperRef = useRef<any>(null)
  const {
    currentStep = 0,
    stepsLength = 1,
    stepsTitleList,
    prevStep,
    nextStep,
  } = useContext(FormContext)
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const { values, errors } = useFormikContext<any>()

  const mappedFields = useMemo(
    () =>
      (Array.isArray(children) ? children : [children]).reduce(
        (acc, item) => {
          if (item.props.name && !item.props.children) {
            acc[item.props.name] = true
          }
          if (
            item.props.name &&
            item.props.required &&
            !item.props.visibleCondition
          ) {
            acc.requiredFields[item.props.name] = true
          }
          if (item.props && item.props.name === 'agreements') {
            acc.requiredFields[item.props.name] = true
          }

          if (item.props.children) {
            const mappedChildrens = Array.isArray(item.props.children)
              ? item.props.children
              : [item.props.children]

            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            mappedChildrens.forEach((child: any) => {
              if (
                child.props.name &&
                child.props.required &&
                !child.props.visibleCondition
              ) {
                if (item.props.type === 'checkboxGroup') {
                  !acc.requiredFields[item.props.name] &&
                    (acc.requiredFields[item.props.name] = {})
                  acc.requiredFields[item.props.name][child.props.name] = true
                  return
                }

                acc.requiredFields[child.props.name] = true
              }
              if (child.props.name) {
                acc[child.props.name] = true
              }
            })
          }

          return acc
        },
        { requiredFields: {} }
      ),
    [children]
  )

  const [nextButtonDisabled, setNextButtonDisabled] = useState(
    JSON.stringify(mappedFields) !== JSON.stringify({})
  )
  const lastStepIndex = stepsLength ? stepsLength - 1 : 0

  useEffect(() => {
    let hasError

    Object.keys(mappedFields.requiredFields).some((key: string) => {
      if (!values[key] || values[key] === '' || errors[key]) {
        hasError = true
        return true
      }

      if (key === 'agreements' && errors[key]) {
        hasError = true
        return true
      }

      return false
    })

    if (!hasError) {
      Object.keys(mappedFields).some((key: string) => {
        if (errors[key]) {
          hasError = true
          return true
        }

        return false
      })
    }

    if (hasError) {
      setNextButtonDisabled(true)
    }

    if (!hasError) {
      setNextButtonDisabled(false)
    }
  }, [values, errors, mappedFields, children])

  const handleClick = () => {
    if (onStepComplete !== undefined) {
      onStepComplete()
    }

    if (nextStep !== undefined) {
      nextStep()
    }
  }

  useEffect(() => {
    if (stepIndex === currentStep) {
      wrapperRef.current !== null &&
        window.scrollTo(0, wrapperRef.current.offsetTop)
    }
  }, [currentStep, stepIndex])

  return (
    <Wrapper ref={wrapperRef} visible={stepIndex === currentStep}>
      <StepHeaderWrapper>
        {currentStep !== 0 && (
          <StepHeader>
            {stepsTitleList && stepsTitleList[currentStep - 1]}
          </StepHeader>
        )}
        {stepIndex === currentStep && (
          <StepHeader activeStep>
            {stepsTitleList && stepsTitleList[currentStep]}
          </StepHeader>
        )}
        {currentStep === 0 && stepsLength > 1 && (
          <StepHeader>
            {stepsTitleList && stepsTitleList[currentStep + 1]}
          </StepHeader>
        )}
      </StepHeaderWrapper>
      {stepIndex === currentStep && children}
      <ButtonsWrapper isFirstStep={currentStep === 0}>
        {currentStep !== 0 && (
          <Button
            type="button"
            variant="secondary"
            text="Cofnij"
            onClick={prevStep}
          />
        )}
        {currentStep !== lastStepIndex && (
          <Button
            disabled={nextButtonDisabled}
            type="button"
            text="Dalej"
            onClick={handleClick}
          />
        )}
        {currentStep === lastStepIndex && (
          <Button disabled={nextButtonDisabled} type="submit" text="WyÅ›lij" />
        )}
      </ButtonsWrapper>
    </Wrapper>
  )
}

export default Step
