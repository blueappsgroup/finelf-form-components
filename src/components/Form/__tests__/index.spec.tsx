import * as React from 'react'
import { render, act, waitFor, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import FirstNameField from '../../FormInput/FirstNameField'
import Step from '../../Step'
import LastNameField from '../../FormInput/LastNameField'
import Form from '../index'

describe('base <Form />', () => {
  const onSubmit = jest.fn()
  const stepsLength = 2
  const stepsTitles = ['1. Podstawowe dane', '2. Szczegółowe dane']
  window.scrollTo = jest.fn()
  const setupWrapper = async ({
    formId = 'testForm',
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  }) => {
    let wrapper
    await act(
      async () =>
        (wrapper = render(
          <Form
            id={formId}
            onSubmit={onSubmit}
            stepsLength={stepsLength}
            stepsTitles={stepsTitles}
          >
            <Step stepIndex={0}>
              <FirstNameField name="fName" required />
            </Step>
            <Step stepIndex={1}>
              <LastNameField name="lName" required />
            </Step>
          </Form>
        ))
    )
    return wrapper
  }

  const setupSingleFormWrapper = async ({
    formId = 'testForm',
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  }) => {
    let wrapper
    await act(
      async () =>
        (wrapper = render(
          <Form id={formId} onSubmit={onSubmit}>
            <FirstNameField name="fName" required />
          </Form>
        ))
    )
    return wrapper
  }

  it('without matches snapshot', async () => {
    const wrapper = await setupSingleFormWrapper({})

    expect(wrapper).toMatchSnapshot()
  })

  it('buttons disabled if fields are filled improperly form without steps', async () => {
    const wrapper = await setupSingleFormWrapper({})

    userEvent.type(screen.getByRole('textbox'), '123')

    expect(screen.getByRole('button')).toBeDisabled()
    expect(wrapper).toMatchSnapshot()
  })

  it('buttons enabled fields are filled properly form without steps', async () => {
    await setupSingleFormWrapper({})

    userEvent.type(screen.getByRole('textbox'), 'John')

    await waitFor(() => {
      expect(screen.getByText('Wyślij')).not.toBeDisabled()
    })
  })

  it('multi step matches snapshot', async () => {
    const wrapper = await setupWrapper({})

    expect(wrapper).toMatchSnapshot()
  })

  it('buttons disabled if required fields are not filled', async () => {
    const wrapper = await setupWrapper({})

    expect(screen.getByRole('button')).toBeDisabled()
    expect(wrapper).toMatchSnapshot()
  })

  it('buttons disabled if fields are filled improperly', async () => {
    const wrapper = await setupWrapper({})

    userEvent.type(screen.getByRole('textbox'), '123')

    expect(screen.getByRole('button')).toBeDisabled()
    expect(wrapper).toMatchSnapshot()
  })

  it('it should through steps properly', async () => {
    await setupWrapper({})

    userEvent.type(screen.getByRole('textbox'), 'John')

    await waitFor(() => {
      expect(screen.getByText('Dalej')).not.toBeDisabled()
      userEvent.click(screen.getByText('Dalej'))
    })

    await waitFor(() => {
      expect(screen.getByText('Wyślij')).toBeTruthy()
    })
  })

  it('it should go next and back in steps properly', async () => {
    await setupWrapper({})

    userEvent.type(screen.getByRole('textbox'), 'John')

    await waitFor(() => {
      expect(screen.getByText('Dalej')).not.toBeDisabled()
      userEvent.click(screen.getByText('Dalej'))
    })

    await waitFor(() => {
      expect(screen.getByText('Cofnij')).toBeTruthy()
      userEvent.click(screen.getByText('Cofnij'))
    })

    await waitFor(() => {
      expect(screen.getByText('Dalej')).toBeTruthy()
    })
  })
})
