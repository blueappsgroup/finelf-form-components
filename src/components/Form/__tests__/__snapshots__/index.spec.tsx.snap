// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base <Form /> buttons disabled if fields are filled improperly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-giIncl jPAApu"
        id="0"
      >
        <div
          class="sc-ezrdKe gBFqDK"
        >
          <div
            class="sc-kLgntA gqMeOj"
          >
            <div
              class="sc-dmlrTW ccRtsS"
            >
              1. Podstawowe dane
            </div>
            <div
              class="sc-dmlrTW bVFKcM"
            >
              2. Szczegółowe dane
            </div>
          </div>
          <div
            class="sc-hKgILt bckkbl"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-dIUggk hEHKqS"
            >
              <input
                class="sc-pFZIQ kIcSeo"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value="123"
              />
            </div>
          </div>
          <div
            class="sc-bYEvPH iVCcgK"
          >
            <button
              class="sc-hHftDr clzDPz"
              disabled=""
              form="0"
              type="submit"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-giIncl jPAApu"
      id="0"
    >
      <div
        class="sc-ezrdKe gBFqDK"
      >
        <div
          class="sc-kLgntA gqMeOj"
        >
          <div
            class="sc-dmlrTW ccRtsS"
          >
            1. Podstawowe dane
          </div>
          <div
            class="sc-dmlrTW bVFKcM"
          >
            2. Szczegółowe dane
          </div>
        </div>
        <div
          class="sc-hKgILt bckkbl"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-dIUggk hEHKqS"
          >
            <input
              class="sc-pFZIQ kIcSeo"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value="123"
            />
          </div>
        </div>
        <div
          class="sc-bYEvPH iVCcgK"
        >
          <button
            class="sc-hHftDr clzDPz"
            disabled=""
            form="0"
            type="submit"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Form /> buttons disabled if fields are filled improperly form without steps 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-giIncl jPAApu"
        id="testForm"
      >
        <div
          class="sc-hKgILt bckkbl"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-dIUggk hEHKqS"
          >
            <input
              class="sc-pFZIQ kIcSeo"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value="123"
            />
          </div>
        </div>
        <button
          class="sc-hHftDr clzDPz"
          disabled=""
          type="submit"
        >
          Wyślij
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-giIncl jPAApu"
      id="testForm"
    >
      <div
        class="sc-hKgILt bckkbl"
      >
        <label
          for="fName"
        >
          Imię*
        </label>
        <div
          class="sc-dIUggk hEHKqS"
        >
          <input
            class="sc-pFZIQ kIcSeo"
            label="Imię"
            name="fName"
            placeholder="Imię*"
            required=""
            value="123"
          />
        </div>
      </div>
      <button
        class="sc-hHftDr clzDPz"
        disabled=""
        type="submit"
      >
        Wyślij
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Form /> buttons disabled if required fields are not filled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-giIncl jPAApu"
        id="0"
      >
        <div
          class="sc-ezrdKe gBFqDK"
        >
          <div
            class="sc-kLgntA gqMeOj"
          >
            <div
              class="sc-dmlrTW ccRtsS"
            >
              1. Podstawowe dane
            </div>
            <div
              class="sc-dmlrTW bVFKcM"
            >
              2. Szczegółowe dane
            </div>
          </div>
          <div
            class="sc-hKgILt bckkbl"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-dIUggk hEHKqS"
            >
              <input
                class="sc-pFZIQ kIcSeo"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value=""
              />
            </div>
          </div>
          <div
            class="sc-bYEvPH iVCcgK"
          >
            <button
              class="sc-hHftDr clzDPz"
              disabled=""
              form="0"
              type="submit"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-giIncl jPAApu"
      id="0"
    >
      <div
        class="sc-ezrdKe gBFqDK"
      >
        <div
          class="sc-kLgntA gqMeOj"
        >
          <div
            class="sc-dmlrTW ccRtsS"
          >
            1. Podstawowe dane
          </div>
          <div
            class="sc-dmlrTW bVFKcM"
          >
            2. Szczegółowe dane
          </div>
        </div>
        <div
          class="sc-hKgILt bckkbl"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-dIUggk hEHKqS"
          >
            <input
              class="sc-pFZIQ kIcSeo"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value=""
            />
          </div>
        </div>
        <div
          class="sc-bYEvPH iVCcgK"
        >
          <button
            class="sc-hHftDr clzDPz"
            disabled=""
            form="0"
            type="submit"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Form /> multi step matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-giIncl jPAApu"
        id="0"
      >
        <div
          class="sc-ezrdKe gBFqDK"
        >
          <div
            class="sc-kLgntA gqMeOj"
          >
            <div
              class="sc-dmlrTW ccRtsS"
            >
              1. Podstawowe dane
            </div>
            <div
              class="sc-dmlrTW bVFKcM"
            >
              2. Szczegółowe dane
            </div>
          </div>
          <div
            class="sc-hKgILt bckkbl"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-dIUggk hEHKqS"
            >
              <input
                class="sc-pFZIQ kIcSeo"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value=""
              />
            </div>
          </div>
          <div
            class="sc-bYEvPH iVCcgK"
          >
            <button
              class="sc-hHftDr clzDPz"
              disabled=""
              form="0"
              type="submit"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-giIncl jPAApu"
      id="0"
    >
      <div
        class="sc-ezrdKe gBFqDK"
      >
        <div
          class="sc-kLgntA gqMeOj"
        >
          <div
            class="sc-dmlrTW ccRtsS"
          >
            1. Podstawowe dane
          </div>
          <div
            class="sc-dmlrTW bVFKcM"
          >
            2. Szczegółowe dane
          </div>
        </div>
        <div
          class="sc-hKgILt bckkbl"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-dIUggk hEHKqS"
          >
            <input
              class="sc-pFZIQ kIcSeo"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value=""
            />
          </div>
        </div>
        <div
          class="sc-bYEvPH iVCcgK"
        >
          <button
            class="sc-hHftDr clzDPz"
            disabled=""
            form="0"
            type="submit"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Form /> without matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-giIncl jPAApu"
        id="testForm"
      >
        <div
          class="sc-hKgILt bckkbl"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-dIUggk hEHKqS"
          >
            <input
              class="sc-pFZIQ kIcSeo"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value=""
            />
          </div>
        </div>
        <button
          class="sc-hHftDr clzDPz"
          disabled=""
          type="submit"
        >
          Wyślij
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-giIncl jPAApu"
      id="testForm"
    >
      <div
        class="sc-hKgILt bckkbl"
      >
        <label
          for="fName"
        >
          Imię*
        </label>
        <div
          class="sc-dIUggk hEHKqS"
        >
          <input
            class="sc-pFZIQ kIcSeo"
            label="Imię"
            name="fName"
            placeholder="Imię*"
            required=""
            value=""
          />
        </div>
      </div>
      <button
        class="sc-hHftDr clzDPz"
        disabled=""
        type="submit"
      >
        Wyślij
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
